services/job.cpp

```diff
@@ -28,13 +28,14 @@
 
 using namespace std;
 
-list<string> CompileJob::flags(Argument_Type argumentType) const
```

Comment:
I hate to lose type safety.

Reply:
Yes, I felt so too and though about it for some time. But I could not find a good solution. 
I could revert the ArgumentType enum to not use flags and implement customized for loops for remoteAndRestFlags() and restFlags(). That would cause some code duplication, but keep the type safety.
Should I do that instead, or do you have an alternative solution?